morango/sync/syncsession.py

```diff
@@ -485,20 +584,22 @@ def _pull_records(self, callback=None):
                     "Specified TransferSession does not match this SyncClient's current TransferSession."
                 )
 
-            validate_and_create_buffer_data(data, self.current_transfer_session)
+            validate_and_create_buffer_data(
+                data, self.current_transfer_session, self.sync_connection
```

Comment:
`connection=self.sync_connection` could be clearer here, since it's an optional/named argument